### PR TITLE
Cleaned up and Modularized Code that was Repetitive

### PR DESCRIPTION
**Pull request by ekatiyar**
Seems to be working using python program, but please test using Iphone before merge

### CODE REVIEW

File: software/apps/romi_control/states.c [hunk 1]
```diff
@@ -62,42 +52,32 @@ void transition_out(inputs_t input_state, system_state_t* curr_state, states old
...
-			if (input_state.has_recently_connected) {
+			if (input_state->has_recently_connected) {
           		curr_state->state = WAITING;
-				break;
         	} else {
           		output.left_speed = 0;
           		output.right_speed = 0;
           		curr_state->state = OFF;
-				break;
         	}
```
Comment (aagarwal1999) [inline software/apps/romi_control/states.c:69]: Needs to be a break here.

File: software/apps/romi_control/states.c [hunk 2]
```diff
@@ -106,26 +86,28 @@ outputs_t transition(inputs_t input_state, system_state_t* curr_state) {
...
 	        float diff = angle_modulo(curr_state->turn_angle - curr_state->curr_orientation_angle);
-	        if (input_state.button_pressed || !input_state.has_recently_connected) {
+	        if (input_state->button_pressed || !input_state->has_recently_connected) {
 	          	curr_state->state = OFF;
 	        } else if (fabs(diff) < angle_threshold) {
-	        	//transition out of turning
-	        	//transition into driving
-	        	printf("Transitioning to driving!\n");
-	            curr_state->state = DRIVING;
+				if (curr_state->state == END_TURNING) {
```
Comment (aagarwal1999) [inline software/apps/romi_control/states.c:96]: Not a big fan of having END_TURNING and TURNING in the same case statement. I'd rather have them explicitly be different for simplicity and ease of reading.

File: software/apps/romi_control/states.c [hunk 3]
```diff
@@ -147,38 +129,11 @@ outputs_t transition(inputs_t input_state, system_state_t* curr_state) {
 	          curr_state->state = DRIVING;
 	        }
         	break;
-     	}
```
Comment (aagarwal1999) [inline software/apps/romi_control/states.c:150]: As stated before, I think this part should be reverted.

File: software/apps/romi_control/states.c [hunk 4]
```diff
@@ -404,101 +307,76 @@ outputs_t substate_transition(inputs_t input_state, system_state_t* curr_state){
...
+	}
 	if (curr_state->substate.substate != old_state) {
-     		substate_transition_out(input_state, &(curr_state->substate), old_state);
+     		substate_transition_out(&(curr_state->substate), old_state);
      		substate_transition_in(input_state, &(curr_state->substate));
     }
 	return output;
 }
+
+void translation_control(system_state_t* curr_state, inputs_t* input_state, int dir, int16_t* left_pwr, int16_t* right_pwr) {
```
Comment (aagarwal1999) [inline software/apps/romi_control/states.c:361]: maybe explicitly set `dir` to be an `int8_t`

File: software/apps/romi_control/states.c (hunk 2 above)
Comment (ekatiyar) [reply software/apps/romi_control/states.c:96]: Yeah I thought it'd be controversial. Honestly the amount of repeated code between the two and the shared functionality (the only difference being exit conditions) made me feel the trade off was worth it. Another way to implement this was by using some sort of state variable to switch between exit conditions, but I felt like keeping the END_TURNING switch case using a fall-through was a good compromise b/w cleanliness and readability

File: software/apps/romi_control/states.c (hunk 4 above)
Comment (ekatiyar) [reply software/apps/romi_control/states.c:361]: Is there a benefit beyond reducing the size of the variable by a couple bytes? Wherever it's being used would still require  a conversion of the value from floats to int16_t

**Pull request closed without merging**